enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,428 @@
+  able to reach the machine config server
+
+### Use Case #2 - Extra VLAN, Post Install Configuration
+
+Consider an on-premise environment with workloads that must be able to access
+resources on multiple networks that must be directly connected to the cluster.
+One example could be an isolated network running database services that are not
+routeable from any other network.  Instead, we’ll attach the Nodes to that VLAN
+and limit which pods can access it.
+
+* After cluster installation, configure another VLAN on a network interface
+  that specific workloads will need access to
+
+### Use Case #3 - Static Network Configuration Only
+
+Consider a lab environment where DHCP services either aren’t available at all,
+or the person trying out an OpenShift cluster does not have access to it to
+configure DHCP address reservations.  As long as the cluster administrator has
+a pool of addresses they are allowed to assign manually, a cluster with fully
+static network configuration would allow them to install OpenShift without
+needing to work with the network administrator on DHCP server configuration.
+
+* Must be able to override default RHCOS behavior of requiring DHCP to be
+  present on at least one interface
+* Must be able to configure static network configuration for every interface
+  used by the cluster
+
+
+## Existing Configuration Interfaces
+
+### Summary
+
+First, here’s a summary of how an administrator may configure host networking
+in a cluster.  More details about each interface are in the following sections.
+
+#### First Boot
+
+When an RHCOS host first boots, the initramfs environment must be able to reach
+the [Machine Config
+Server](https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigServer.md)
+for Ignition to succeed.  Ignition will attempt to reach the Machine Config
+Server via an included URL. By default, dracut will attempt to do DHCPv4 and
+DHCPv6 on all interfaces and wait for one of them to work.  To change this
+behavior, an admin can:
+
+* Change the `ip=<...>` kernel argument(s) via PXE configuration or at the grub
+  command line during boot
+* If at least one network has DHCP to allow dracut to succeed, custom ignition
+  additions can configure additional networking
+* If using the RHCOS live ISO/PXE ramdisk environment with the
+  coreos-installer, use the `nmtui` Network Manager text UI to configure
+  networking before proceeding
+
+#### Further Network Customization
+
+Beyond the minimum needed to bootstrap the cluster, further host network
+configuration customizations can be applied in the following ways:
+
+* Customize the ignition configuration at install time to include more
+  NetworkManager configuration files
+* Create `MachineConfig` resources with NetworkManager configuration files and
+  have them applied by the installer at install time, or apply them to the
+  cluster via the API any time post-install
```

Comment:
Should nmstate be mentioned here, or only in the future improvements sections /cc @celebdor

Reply:
Just future, since that's not something included yet.  It would go here whenever it's integrated.